zips/zip-0233.md

```diff
@@ -19,224 +20,103 @@ The key words "MUST", "SHOULD", "SHOULD NOT", "MAY", "RECOMMENDED", "OPTIONAL",
-upgrade. For example, NU5 supports both v4 and v5 transaction formats, for both
-coinbase and non-coinbase transactions.
+We add a new field to the common transaction fields [3]:
 
-### `zsfDeposit` fields in transactions
+    burnAmount : uint64 [zatoshi]
 
-Each transaction can dedicate some of its excess funds to the ZSF, and the
-remainder becomes the miner fee, with any excess miner fee/reward going to the
-ZSF
+This field specifies how much of the input value for the transaction should be
+burned.
 
-This is achieved by adding a new field to the common transaction fields
-[#zip-0225-transaction-format]:
+**Note:** Older transaction versions can continue to be supported after a
+network upgrade, but burning is not possible for these transactions. For
+example, NU5 supports both v4 and v5 transaction formats, for both coinbase and
+non-coinbase transactions.
 
-- `zsfDeposit : u64 [zatoshi]`
+## Consensus Rule Changes
 
-The `ZSF_BALANCE[H]` for a block at height `H` can be calculated given a value
-of `ZSF_BALANCE[H-1]` and the set of transactions contained in that block.
-First, the `ZSF_DEPOSIT[H]` is calculated based solely on `ZSF_BALANCE[H-1]`.
-This is subtracted from the previous block's balance to be distributed as part
-of the block reward. Second, the sum of all the `ZSF_DEPOSIT` fields of all
-transactions in the block is added to the balance.
+The burned value must now be considered when calculating the total output value
+of a transaction. It should be treated similarly to a transparent output,
+except that there is no way for this value to be used as an input in a future
+transaction.
 
+## Transaction Format Requirements
 
-### Non-Coinbase Transactions
-
-If the `ZSF_DEPOSIT` field is not present in an older transaction version, it
-is defined to be zero for non-coinbase transactions.
-
-#### Consensus Rule Changes
-
-- Transparent inputs to a transaction insert value into a transparent
-  transaction value pool associated with the transaction. Transparent outputs
-  **and sustainability fund deposits** remove value from this pool.
-
-### Coinbase Transactions
-
-Any excess miner fee/reward is sent to the ZSF.
-
-In new blocks, this deposit is tracked via the `ZSF_DEPOSIT` field in coinbase
-transactions. 
-
-If the `ZSF_DEPOSIT` field is not present in a coinbase transaction with an
-older transaction version, it is defined as the value of any unspent miner fee
-and miner reward. This re-defines these previously unspendable funds as ZSF
-deposits.
-
-#### Consensus Rule Changes
-
--  The remaining value in the transparent transaction value pool of a coinbase
-   transaction is **deposited in the sustainability fund**.
-
-### `ZSF_DEPOSIT` Requirements
-
-- ZIP 230 [3] must be updated to include `ZSF_DEPOSIT`.
-- ZIP 244 [4] must be updated as well to include `ZSF_DEPOSIT`.
+* ZIP 230 [3] must be updated to include the new `burnAmount` field.
+* ZIP 244 [4] must be updated as well to include the new `burnAmount` field.
```

Comment:
This depends on whether the v6 sighash changes are done as a modification to ZIP 244 or as a new ZIP.
```suggestion
* The `burnAmount` field must be taken account of in sighash computation
  (either in ZIP 244 [4] or in a new ZIP that specifies the sighash algorithm
  for the new transaction version).
```

Reply:
Please follow the example in https://zips.z.cash/draft-ecc-explicit-fees to specify the changes to the transaction format.

Reply:
I've tried to apply this in this commit: https://github.com/zcash/zips/pull/913/commits/86c35a9c5a3924d7eb2f9b2b8148aea86d94cdaf.